File: Dev_Lab Unofficial/Python/หวนคือสู่วานร.py
boxra = float(input())
handsize = float(input())
nut = float(input())
nutsize = float(input())
getnut = 0

while handsize < boxra and nut != 0:
  handsize += nutsize
  getnut += 1
  nut -= 1
if handsize > boxra:
  print(getnut -1)
else:
  print(getnut)

File: Dev_Lab Unofficial/C++/หวนคือสู่วานร.cpp
#include <iostream>
using namespace std;


int main()
{
	double boxra,hand,nut,hand_nut,collect;
  collect = 0;
  cin >> boxra >> hand >> nut >> hand_nut;
  
  while(hand < boxra && nut != 0){
    nut -=1;
    hand += hand_nut;
    if(hand > boxra){
      break;
    }
    else{
      collect +=1;
    }
  }
  cout << collect;
	return 0;
}